Algo.h

```diff
@@ -294,7 +385,8 @@ class MMP
                 // invert wale pains???
 
                 for (auto &e : ((Vertex *)(source->p))->edges) {
-                    Interval ii(0, 0, 0, e->length(), 0, NULL, e);
+                    //check for invert
+                    Interval ii(0, 0, 0, e->length(), 0, NULL, e, false);
```

Comment:
We will check the convention and pass invert according when we are creating new intervals. Once, an interval is formed, assume it's according to the convention